app/assets/javascripts/services/miq_service.js

```diff
@@ -158,4 +158,10 @@ ManageIQ.angular.app.service('miqService', ['$timeout', '$document', '$q', 'API'
       miqService.sparkleOff();
     }
   };
+
+  this.resetData = function(vm, hshName, scope) {
+    vm[hshName] = angular.copy( vm.modelCopy );
+    scope.angularForm.$setPristine(true);
```

Comment:
Passing in scope only to get to the form object and hardcoding the name..  maybe we should just pass `$scope.angularForm` directly?